service/src/main/kotlin/fi/espoo/evaka/messaging/Message.kt

```diff
@@ -48,16 +50,52 @@ data class MessageThreadStub(
     val type: MessageType,
     val title: String,
     val urgent: Boolean,
+    val sensitive: Boolean,
     val isCopy: Boolean
 )
 
+enum class MessageThreadType {
+    REDACTED_MESSAGE_THREAD,
+    MESSAGE_THREAD
+}
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+sealed class CitizenMessageThread(val type: MessageThreadType) {
+    abstract val id: MessageThreadId
+    abstract val urgent: Boolean
+    abstract val children: List<MessageChild>
+
+    @JsonTypeName("REDACTED_MESSAGE_THREAD")
```

Comment:
to be replaced with The new way (TM)

Reply:
I'll test this if it works without the proposed typed Paged system!

Reply:
@akheron tested this and this still would need the typed Paged system in order to remove the enum + type field + type parameter.